SDRpp crashes after signal lock

When locking onto any tetra signal SDRpp crashes straight away.
This is the status of the module, when the crash occurs.
![img](https://github.com/user-attachments/assets/c44836b3-c274-444a-bf94-d14ae06e6630)

Module was installed on arch through the AUR. for hardware a hackrf was used.

Ther are some warnings while compiling. related?

```
==> Making package: sdrpp-tetra-demodulator-git r18.9e3a757-1 (Wed Dec 11 21:38:33 2024)
==> Checking runtime dependencies...
==> Checking buildtime dependencies...
==> Retrieving sources...
  -> Updating sdrpp-tetra-demodulator git repo...
==> Validating source files with md5sums...
    sdrpp-tetra-demodulator ... Skipped
==> Removing existing $srcdir/ directory...
==> Extracting sources...
  -> Creating working copy of sdrpp-tetra-demodulator git repo...
Cloning into 'sdrpp-tetra-demodulator'...
done.
==> Starting pkgver()...
==> Updated version: sdrpp-tetra-demodulator-git r100.d643db5-1
==> Sources are ready.
==> Making package: sdrpp-tetra-demodulator-git r100.d643db5-1 (Wed Dec 11 21:38:35 2024)
==> Checking runtime dependencies...
==> Checking buildtime dependencies...
==> WARNING: Using existing $srcdir/ tree
==> Starting pkgver()...
==> Starting build()...
Deleting /home/damige/.cache/yay/sdrpp-tetra-demodulator-git/src/sdrpp-tetra-demodulator/src/decoder/etsi_codec-patches/../codec ...
Creating /home/damige/.cache/yay/sdrpp-tetra-demodulator-git/src/sdrpp-tetra-demodulator/src/decoder/etsi_codec-patches/../codec ...
Downloading http://www.etsi.org/deliver/etsi_en/300300_300399/30039502/01.03.01_60/en_30039502v010301p0.zip ...
Checking MD5SUM ...
Unpacking ZIP ...
Archive:  /home/damige/.cache/yay/sdrpp-tetra-demodulator-git/src/sdrpp-tetra-demodulator/src/decoder/etsi_codec-patches/etsi_tetra_codec.zip
  inflating: c-word/c_word.win       
  inflating: c-word/c_word_a.doc     
  inflating: c-word/c_word_b.doc     
  inflating: c-word/c_word_c.doc     
  inflating: c-word/c_word_d.doc     
  inflating: c-word/c_word_e.doc     
  inflating: c-word/c_word_f.doc     
  inflating: amr-code/arrays.h       
  inflating: amr-code/arrays.tab     
  inflating: amr-code/arrays_amr475.tab  
  inflating: amr-code/arrays_tetra.tab  
  inflating: amr-code/ccoder.c       
  inflating: amr-code/ccod_tet.c     
  inflating: amr-code/cdecoder.c     
  inflating: amr-code/cdec_tet.c     
  inflating: amr-code/channel.h      
  inflating: amr-code/const.tab      
  inflating: amr-code/const_amr475.tab  
  inflating: amr-code/const_tetra.tab  
  inflating: amr-code/globals.c      
  inflating: amr-code/globals.h      
  inflating: amr-code/init_params.c  
  inflating: amr-code/makefile       
  inflating: amr-code/source.h       
  inflating: amr-code/sub_cc.c       
  inflating: amr-code/sub_cd.c       
  inflating: amr-code/tetra_op.c     
  inflating: c-code/arrays.tab       
  inflating: c-code/ccoder.c         
  inflating: c-code/ccod_tet.c       
  inflating: c-code/cdecoder.c       
  inflating: c-code/cdec_tet.c       
  inflating: c-code/channel.h        
  inflating: c-code/clsp_334.tab     
  inflating: c-code/const.tab        
  inflating: c-code/ener_qua.tab     
  inflating: c-code/fbas_tet.c       
  inflating: c-code/fexp_tet.c       
  inflating: c-code/fmat_tet.c       
  inflating: c-code/grid.tab         
  inflating: c-code/inv_sqrt.tab     
  inflating: c-code/lag_wind.tab     
  inflating: c-code/log2.tab         
  inflating: c-code/makefile         
  inflating: c-code/pow2.tab         
  inflating: c-code/scoder.c         
  inflating: c-code/scod_tet.c       
  inflating: c-code/sdecoder.c       
  inflating: c-code/sdec_tet.c       
  inflating: c-code/source.h         
  inflating: c-code/sub_cc.c         
  inflating: c-code/sub_cd.c         
  inflating: c-code/sub_dsp.c        
  inflating: c-code/sub_sc_d.c       
  inflating: c-code/tetra_op.c       
  inflating: c-code/window.tab       
Contents of /home/damige/.cache/yay/sdrpp-tetra-demodulator-git/src/sdrpp-tetra-demodulator/src/decoder/etsi_codec-patches/../codec:
total 20K
drwxr-xr-x 5 damige damige 4.0K Dec 11 21:38 .
drwxr-xr-x 5 damige damige 4.0K Dec 11 21:38 ..
drwxr-xr-x 2 damige damige 4.0K Dec 11 21:38 amr-code
drwxr-xr-x 2 damige damige 4.0K Dec 11 21:38 c-code
drwxr-xr-x 2 damige damige 4.0K Dec 11 21:38 c-word
Applying Patches ...
total 404K
drwxr-xr-x 2 damige damige 4.0K Dec 11 21:38 .
drwxr-xr-x 5 damige damige 4.0K Dec 11 21:38 ..
-rw-r--r-- 1 damige damige 1.4K Dec 11 21:38 README
-rwxr-xr-x 1 damige damige  988 Dec 11 21:38 download_and_patch.sh
-rw-r--r-- 1 damige damige 353K Dec 11 21:38 etsi_tetra_codec.zip
-rw-r--r-- 1 damige damige 1.2K Dec 11 21:38 filename-case.patch
-rw-r--r-- 1 damige damige  836 Dec 11 21:38 fix_64bit.patch
-rw-r--r-- 1 damige damige 5.2K Dec 11 21:38 log_stderr.patch
-rw-r--r-- 1 damige damige 1.5K Dec 11 21:38 makefile-cleanups.patch
-rw-r--r-- 1 damige damige 6.6K Dec 11 21:38 round_private.patch
-rw-r--r-- 1 damige damige   97 Dec 11 21:38 series
makefile-cleanups.patch
fix_64bit.patch
round_private.patch
filename-case.patch
log_stderr.patch
=> Applying patch '/home/damige/.cache/yay/sdrpp-tetra-demodulator-git/src/sdrpp-tetra-demodulator/src/decoder/etsi_codec-patches/makefile-cleanups.patch'...
patching file amr-code/makefile
patching file c-code/makefile
=> Applying patch '/home/damige/.cache/yay/sdrpp-tetra-demodulator-git/src/sdrpp-tetra-demodulator/src/decoder/etsi_codec-patches/fix_64bit.patch'...
patching file amr-code/source.h
patching file c-code/source.h
=> Applying patch '/home/damige/.cache/yay/sdrpp-tetra-demodulator-git/src/sdrpp-tetra-demodulator/src/decoder/etsi_codec-patches/round_private.patch'...
patching file amr-code/channel.h
patching file amr-code/source.h
Hunk #1 succeeded at 165 (offset 1 line).
patching file amr-code/tetra_op.c
patching file c-code/channel.h
patching file c-code/source.h
Hunk #1 succeeded at 165 (offset 1 line).
patching file c-code/sub_dsp.c
patching file c-code/sub_sc_d.c
patching file c-code/tetra_op.c
=> Applying patch '/home/damige/.cache/yay/sdrpp-tetra-demodulator-git/src/sdrpp-tetra-demodulator/src/decoder/etsi_codec-patches/filename-case.patch'...
patching file amr-code/init_params.c
=> Applying patch '/home/damige/.cache/yay/sdrpp-tetra-demodulator-git/src/sdrpp-tetra-demodulator/src/decoder/etsi_codec-patches/log_stderr.patch'...
patching file c-code/cdecoder.c
patching file c-code/sdecoder.c
Done!
-- The C compiler identification is GNU 14.2.1
-- The CXX compiler identification is GNU 14.2.1
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Configuring done (0.5s)
-- Generating done (0.0s)
-- Build files have been written to: /home/damige/.cache/yay/sdrpp-tetra-demodulator-git/src/sdrpp-tetra-demodulator/build
[  6%] Building C object CMakeFiles/tetra_demodulator.dir/src/decoder/codec/c-code/cdec_tet.c.o
[  6%] Building C object CMakeFiles/tetra_demodulator.dir/src/decoder/codec/c-code/fmat_tet.c.o
[ 11%] Building C object CMakeFiles/tetra_demodulator.dir/src/decoder/codec/c-code/fbas_tet.c.o
[ 11%] Building C object CMakeFiles/tetra_demodulator.dir/src/decoder/codec/c-code/fexp_tet.c.o
[ 13%] Building C object CMakeFiles/tetra_demodulator.dir/src/decoder/codec/c-code/sub_dsp.c.o
[ 15%] Building C object CMakeFiles/tetra_demodulator.dir/src/decoder/codec/c-code/sub_cd.c.o
[ 15%] Building C object CMakeFiles/tetra_demodulator.dir/src/decoder/codec/c-code/sdec_tet.c.o
[ 17%] Building C object CMakeFiles/tetra_demodulator.dir/src/decoder/codec/c-code/tetra_op.c.o
[ 20%] Building C object CMakeFiles/tetra_demodulator.dir/src/decoder/codec/c-code/sub_sc_d.c.o
[ 22%] Building C object CMakeFiles/tetra_demodulator.dir/src/decoder/src/crypto/hurdle.c.o
[ 24%] Building C object CMakeFiles/tetra_demodulator.dir/src/decoder/src/crypto/taa1.c.o
[ 28%] Building C object CMakeFiles/tetra_demodulator.dir/src/decoder/src/crypto/tea2.c.o
[ 28%] Building C object CMakeFiles/tetra_demodulator.dir/src/decoder/src/crypto/tea1.c.o
[ 31%] Building C object CMakeFiles/tetra_demodulator.dir/src/decoder/src/crypto/tea3.c.o
[ 33%] Building C object CMakeFiles/tetra_demodulator.dir/src/decoder/src/crypto/tetra_crypto.c.o
cc1: warning: command-line option ‘-std=c++17’ is valid for C++/ObjC++ but not for C
cc1: warning: command-line option ‘-std=c++17’ is valid for C++/ObjC++ but not for C
cc1: warning: command-line option ‘-std=c++17’ is valid for C++/ObjC++ but not for C
cc1: warning: command-line option ‘-std=c++17’ is valid for C++/ObjC++ but not for C
cc1: warning: command-line option ‘-std=c++17’ is valid for C++/ObjC++ but not for C
cc1: warning: command-line option ‘-std=c++17’ is valid for C++/ObjC++ but not for C
cc1: warning: command-line option ‘-std=c++17’ is valid for C++/ObjC++ but not for C
cc1: warning: command-line option ‘-std=c++17’ is valid for C++/ObjC++ but not for C
cc1: warning: command-line option ‘-std=c++17’ is valid for C++/ObjC++ but not for C
cc1: warning: command-line option ‘-std=c++17’ is valid for C++/ObjC++ but not for C
cc1: warning: command-line option ‘-std=c++17’ is valid for C++/ObjC++ but not for C
cc1: warning: command-line option ‘-std=c++17’ is valid for C++/ObjC++ but not for C
cc1: warning: command-line option ‘-std=c++17’ is valid for C++/ObjC++ but not for C
cc1: warning: command-line option ‘-std=c++17’ is valid for C++/ObjC++ but not for C
cc1: warning: command-line option ‘-std=c++17’ is valid for C++/ObjC++ but not for C
[ 37%] Building C object CMakeFiles/tetra_demodulator.dir/src/decoder/src/lower_mac/osmo_conv.c.o
[ 37%] Building C object CMakeFiles/tetra_demodulator.dir/src/decoder/src/lower_mac/crc_simple.c.o
[ 40%] Building C object CMakeFiles/tetra_demodulator.dir/src/decoder/src/lower_mac/tch_reordering.c.o
[ 42%] Building C object CMakeFiles/tetra_demodulator.dir/src/decoder/src/lower_mac/tetra_conv_enc.c.o
cc1: warning: command-line option ‘-std=c++17’ is valid for C++/ObjC++ but not for C
cc1: warning: command-line option ‘-std=c++17’ is valid for C++/ObjC++ but not for C
cc1: warning: command-line option ‘-std=c++17’ is valid for C++/ObjC++ but not for C
[ 44%] Building C object CMakeFiles/tetra_demodulator.dir/src/decoder/src/lower_mac/tetra_interleave.c.o
[ 48%] Building C object CMakeFiles/tetra_demodulator.dir/src/decoder/src/lower_mac/tetra_rm3014.c.o
[ 46%] Building C object CMakeFiles/tetra_demodulator.dir/src/decoder/src/lower_mac/tetra_lower_mac.c.o
cc1: warning: command-line option ‘-std=c++17’ is valid for C++/ObjC++ but not for C
[ 51%] Building C object CMakeFiles/tetra_demodulator.dir/src/decoder/src/lower_mac/tetra_scramb.c.o
cc1: warning: command-line option ‘-std=c++17’ is valid for C++/ObjC++ but not for C
cc1: warning: command-line option ‘-std=c++17’ is valid for C++/ObjC++ but not for C
cc1: warning: command-line option ‘-std=c++17’ is valid for C++/ObjC++ but not for C
cc1: warning: command-line option ‘-std=c++17’ is valid for C++/ObjC++ but not for C
[ 53%] Building C object CMakeFiles/tetra_demodulator.dir/src/decoder/src/lower_mac/viterbi.c.o
[ 55%] Building C object CMakeFiles/tetra_demodulator.dir/src/decoder/src/lower_mac/viterbi_cch.c.o
cc1: warning: command-line option ‘-std=c++17’ is valid for C++/ObjC++ but not for C
cc1: warning: command-line option ‘-std=c++17’ is valid for C++/ObjC++ but not for C
[ 57%] Building C object CMakeFiles/tetra_demodulator.dir/src/decoder/src/lower_mac/viterbi_tch.c.o
[ 62%] Building C object CMakeFiles/tetra_demodulator.dir/src/decoder/src/phy/tetra_burst.c.o
[ 62%] Building C object CMakeFiles/tetra_demodulator.dir/src/decoder/src/phy/tetra_burst_sync.c.o
cc1: warning: command-line option ‘-std=c++17’ is valid for C++/ObjC++ but not for C
cc1: warning: command-line option ‘-std=c++17’ is valid for C++/ObjC++ but not for C
cc1: warning: command-line option ‘-std=c++17’ is valid for C++/ObjC++ but not for C
[ 64%] Building C object CMakeFiles/tetra_demodulator.dir/src/decoder/src/tetra_cmce_pdu.c.o
[ 66%] Building C object CMakeFiles/tetra_demodulator.dir/src/decoder/src/tetra_common.c.o
cc1: warning: command-line option ‘-std=c++17’ is valid for C++/ObjC++ but not for C
cc1: warning: command-line option ‘-std=c++17’ is valid for C++/ObjC++ but not for C
[ 71%] Building C object CMakeFiles/tetra_demodulator.dir/src/decoder/src/tetra_mac_pdu.c.o
[ 71%] Building C object CMakeFiles/tetra_demodulator.dir/src/decoder/src/tetra_mle.c.o
[ 73%] Building C object CMakeFiles/tetra_demodulator.dir/src/decoder/src/tetra_mle_pdu.c.o
[ 75%] Building C object CMakeFiles/tetra_demodulator.dir/src/decoder/src/tetra_mm_pdu.c.o
cc1: warning: command-line option ‘-std=c++17’ is valid for C++/ObjC++ but not for C
cc1: warning: command-line option ‘-std=c++17’ is valid for C++/ObjC++ but not for C
cc1: warning: command-line option ‘-std=c++17’ is valid for C++/ObjC++ but not for C
cc1: warning: command-line option ‘-std=c++17’ is valid for C++/ObjC++ but not for C
[ 77%] Building C object CMakeFiles/tetra_demodulator.dir/src/decoder/src/tetra_tdma.c.o
[ 80%] Building C object CMakeFiles/tetra_demodulator.dir/src/decoder/src/tetra_upper_mac.c.o
[ 82%] Building C object CMakeFiles/tetra_demodulator.dir/src/decoder/src/tetra_sndcp_pdu.c.o
cc1: warning: command-line option ‘-std=c++17’ is valid for C++/ObjC++ but not for C
[ 84%] Building CXX object CMakeFiles/tetra_demodulator.dir/src/dsp/bit_unpacker.cpp.o
[ 86%] Building CXX object CMakeFiles/tetra_demodulator.dir/src/dsp/complex_fd.cpp.o
cc1: warning: command-line option ‘-std=c++17’ is valid for C++/ObjC++ but not for C
cc1: warning: command-line option ‘-std=c++17’ is valid for C++/ObjC++ but not for C
[ 91%] Building CXX object CMakeFiles/tetra_demodulator.dir/src/dsp/dqpsk_sym_extr.cpp.o
[ 91%] Building CXX object CMakeFiles/tetra_demodulator.dir/src/dsp/fll.cpp.o
[ 93%] Building CXX object CMakeFiles/tetra_demodulator.dir/src/dsp/pi4dqpsk.cpp.o
[ 95%] Building CXX object CMakeFiles/tetra_demodulator.dir/src/dsp/pi4dqpsk_costas.cpp.o
[ 97%] Building CXX object CMakeFiles/tetra_demodulator.dir/src/main.cpp.o
[100%] Linking CXX shared library tetra_demodulator.so
/usr/src/debug/sdrpp-tetra-demodulator-git/sdrpp-tetra-demodulator/src/decoder/codec/c-code/channel.h:141:8: warning: type of ‘L_deposit_l’ does not match original declaration [-Wlto-type-mismatch]
  141 | Word32 L_deposit_l(Word16 var1);          /* 16 bit var1 -> LSB,  2 */
      |        ^
/usr/src/debug/sdrpp-tetra-demodulator-git/sdrpp-tetra-demodulator/src/decoder/codec/c-code/tetra_op.c:494:8: note: return value type mismatch
  494 | Word32 L_deposit_l(Word16 var1)
      |        ^
/usr/src/debug/sdrpp-tetra-demodulator-git/sdrpp-tetra-demodulator/src/decoder/codec/c-code/tetra_op.c:494:8: note: ‘L_deposit_l’ was previously declared here
/usr/src/debug/sdrpp-tetra-demodulator-git/sdrpp-tetra-demodulator/src/decoder/codec/c-code/tetra_op.c:494:8: note: code may be misoptimized unless ‘-fno-strict-aliasing’ is used
/usr/src/debug/sdrpp-tetra-demodulator-git/sdrpp-tetra-demodulator/src/decoder/codec/c-code/channel.h:126:8: warning: type of ‘extract_h’ does not match original declaration [-Wlto-type-mismatch]
  126 | Word16 extract_h(Word32 L_var1);          /* Extract high,        1 */
      |        ^
/usr/src/debug/sdrpp-tetra-demodulator-git/sdrpp-tetra-demodulator/src/decoder/codec/c-code/tetra_op.c:282:8: note: type mismatch in parameter 1
  282 | Word16 extract_h(Word32 L_var1)
      |        ^
/usr/src/debug/sdrpp-tetra-demodulator-git/sdrpp-tetra-demodulator/src/decoder/codec/c-code/tetra_op.c:282:8: note: ‘extract_h’ was previously declared here
/usr/src/debug/sdrpp-tetra-demodulator-git/sdrpp-tetra-demodulator/src/decoder/codec/c-code/tetra_op.c:282:8: note: code may be misoptimized unless ‘-fno-strict-aliasing’ is used
/usr/src/debug/sdrpp-tetra-demodulator-git/sdrpp-tetra-demodulator/src/decoder/codec/c-code/channel.h:149:8: warning: type of ‘L_shl’ does not match original declaration [-Wlto-type-mismatch]
  149 | Word32 L_shl(Word32 L_var1, Word16 var2); /* Long shift left,     2 */
      |        ^
/usr/src/debug/sdrpp-tetra-demodulator-git/sdrpp-tetra-demodulator/src/decoder/codec/c-code/tetra_op.c:860:8: note: return value type mismatch
  860 | Word32 L_shl(Word32 L_var1, Word16 var2)
      |        ^
/usr/src/debug/sdrpp-tetra-demodulator-git/sdrpp-tetra-demodulator/src/decoder/codec/c-code/tetra_op.c:860:8: note: ‘L_shl’ was previously declared here
/usr/src/debug/sdrpp-tetra-demodulator-git/sdrpp-tetra-demodulator/src/decoder/codec/c-code/tetra_op.c:860:8: note: code may be misoptimized unless ‘-fno-strict-aliasing’ is used
[100%] Built target tetra_demodulator
==> Entering fakeroot environment...
==> Starting package()...
[  2%] Linking CXX shared library tetra_demodulator.so
/usr/src/debug/sdrpp-tetra-demodulator-git/sdrpp-tetra-demodulator/src/decoder/codec/c-code/channel.h:141:8: warning: type of ‘L_deposit_l’ does not match original declaration [-Wlto-type-mismatch]
  141 | Word32 L_deposit_l(Word16 var1);          /* 16 bit var1 -> LSB,  2 */
      |        ^
/usr/src/debug/sdrpp-tetra-demodulator-git/sdrpp-tetra-demodulator/src/decoder/codec/c-code/tetra_op.c:494:8: note: return value type mismatch
  494 | Word32 L_deposit_l(Word16 var1)
      |        ^
/usr/src/debug/sdrpp-tetra-demodulator-git/sdrpp-tetra-demodulator/src/decoder/codec/c-code/tetra_op.c:494:8: note: ‘L_deposit_l’ was previously declared here
/usr/src/debug/sdrpp-tetra-demodulator-git/sdrpp-tetra-demodulator/src/decoder/codec/c-code/tetra_op.c:494:8: note: code may be misoptimized unless ‘-fno-strict-aliasing’ is used
/usr/src/debug/sdrpp-tetra-demodulator-git/sdrpp-tetra-demodulator/src/decoder/codec/c-code/channel.h:126:8: warning: type of ‘extract_h’ does not match original declaration [-Wlto-type-mismatch]
  126 | Word16 extract_h(Word32 L_var1);          /* Extract high,        1 */
      |        ^
/usr/src/debug/sdrpp-tetra-demodulator-git/sdrpp-tetra-demodulator/src/decoder/codec/c-code/tetra_op.c:282:8: note: type mismatch in parameter 1
  282 | Word16 extract_h(Word32 L_var1)
      |        ^
/usr/src/debug/sdrpp-tetra-demodulator-git/sdrpp-tetra-demodulator/src/decoder/codec/c-code/tetra_op.c:282:8: note: ‘extract_h’ was previously declared here
/usr/src/debug/sdrpp-tetra-demodulator-git/sdrpp-tetra-demodulator/src/decoder/codec/c-code/tetra_op.c:282:8: note: code may be misoptimized unless ‘-fno-strict-aliasing’ is used
/usr/src/debug/sdrpp-tetra-demodulator-git/sdrpp-tetra-demodulator/src/decoder/codec/c-code/channel.h:149:8: warning: type of ‘L_shl’ does not match original declaration [-Wlto-type-mismatch]
  149 | Word32 L_shl(Word32 L_var1, Word16 var2); /* Long shift left,     2 */
      |        ^
/usr/src/debug/sdrpp-tetra-demodulator-git/sdrpp-tetra-demodulator/src/decoder/codec/c-code/tetra_op.c:860:8: note: return value type mismatch
  860 | Word32 L_shl(Word32 L_var1, Word16 var2)
      |        ^
/usr/src/debug/sdrpp-tetra-demodulator-git/sdrpp-tetra-demodulator/src/decoder/codec/c-code/tetra_op.c:860:8: note: ‘L_shl’ was previously declared here
/usr/src/debug/sdrpp-tetra-demodulator-git/sdrpp-tetra-demodulator/src/decoder/codec/c-code/tetra_op.c:860:8: note: code may be misoptimized unless ‘-fno-strict-aliasing’ is used
[100%] Built target tetra_demodulator
Install the project...
-- Install configuration: ""
-- Installing: /home/damige/.cache/yay/sdrpp-tetra-demodulator-git/pkg/sdrpp-tetra-demodulator-git/usr/lib/sdrpp/plugins/tetra_demodulator.so
==> Tidying install...
  -> Removing libtool files...
  -> Purging unwanted files...
  -> Removing static library files...
  -> Stripping unneeded symbols from binaries and libraries...
  -> Compressing man and info pages...
==> Checking for packaging issues...
==> Creating package "sdrpp-tetra-demodulator-git"...
  -> Generating .PKGINFO file...
  -> Generating .BUILDINFO file...
  -> Generating .MTREE file...
  -> Compressing package...
==> Creating package "sdrpp-tetra-demodulator-git-debug"...
  -> Generating .PKGINFO file...
  -> Generating .BUILDINFO file...
  -> Generating .MTREE file...
  -> Compressing package...
==> Leaving fakeroot environment.
==> Finished making: sdrpp-tetra-demodulator-git r100.d643db5-1 (Wed Dec 11 21:38:47 2024)
==> Cleaning up...

```

--- a/src/decoder/src/phy/tetra_burst.c
+++ b/src/decoder/src/phy/tetra_burst.c
@@ -292,7 +292,11 @@ int tetra_find_train_seq(const uint8_t *in, unsigned int end_of_in,
 
 	const uint8_t *cur;
 
-	for (cur = in; cur < in + end_of_in; cur++) {
+	/* each iteration reads cur[FILTER_LOOKAHEAD_LEN-1], so stop early enough
+	 * that this lookahead never reads past in[end_of_in-1] -- otherwise we
+	 * run off the end of bitbuf[] into the rest of the struct and beyond
+	 * (heap-buffer-overflow caught by ASan; corrupts burst_cb_priv). */
+	for (cur = in; cur + (FILTER_LOOKAHEAD_LEN-1) < in + end_of_in; cur++) {
 		filter = ((filter << 1) | cur[FILTER_LOOKAHEAD_LEN-1]) & FILTER_LOOKAHEAD_MASK;
 
 		int match = 0;
@@ -345,7 +349,8 @@ void tetra_burst_rx_cb(const uint8_t *burst, unsigned int len, enum tetra_train_
 	uint8_t bbk_buf[NDB_BBK_BITS];
 	uint8_t ndbf_buf[2*NDB_BLK_BITS];
 	struct tetra_mac_state *tms = priv;
-	
+
+	if (!tms || !tms->t_display_st) return; /* decoder state not ready yet */
 	tms->t_display_st->curr_multiframe = t_phy_state.time.mn;
 	tms->t_display_st->curr_frame = t_phy_state.time.fn;
 

--- a/src/decoder/src/phy/tetra_burst_sync.c
+++ b/src/decoder/src/phy/tetra_burst_sync.c
@@ -37,16 +37,26 @@ void tetra_burst_rx_cb(const uint8_t *burst, unsigned int len, enum tetra_train_
 
 static void make_bitbuf_space(struct tetra_rx_state *trs, unsigned int len)
 {
-	unsigned int bitbuf_space = sizeof(trs->bitbuf) - trs->bits_in_buf;
+	unsigned int bitbuf_space;
+
+	/* defensive: if the accounting ever goes out of range, an unsigned
+	 * underflow below would make us skip the shift and let the caller's
+	 * memcpy write past bitbuf[] into the following struct fields
+	 * (e.g. burst_cb_priv). Reset rather than corrupt memory. */
+	if (trs->bits_in_buf > sizeof(trs->bitbuf))
+		trs->bits_in_buf = 0;
+
+	bitbuf_space = sizeof(trs->bitbuf) - trs->bits_in_buf;
 
 	if (bitbuf_space < len) {
 		unsigned int delta = len - bitbuf_space;
+		if (delta > trs->bits_in_buf)
+			delta = trs->bits_in_buf;
 
 		DEBUGP("bitbuf left: %u, shrinking by %u\n", bitbuf_space, delta);
 		memmove(trs->bitbuf, trs->bitbuf + delta, trs->bits_in_buf - delta);
 		trs->bits_in_buf -= delta;
 		trs->bitbuf_start_bitnum += delta;
-		bitbuf_space = sizeof(trs->bitbuf) - trs->bits_in_buf;
 	}
 }
 
@@ -59,6 +69,11 @@ int tetra_burst_sync_in(struct tetra_rx_state *trs, uint8_t *bits, unsigned int 
 	DEBUGP("burst_sync_in: %u bits, state %u\n", len, trs->state);
 
 	/* First: append the data to the bitbuf */
+	if (len > sizeof(trs->bitbuf)) {
+		/* never accept more than the buffer can hold; keep newest bits */
+		bits += len - sizeof(trs->bitbuf);
+		len = sizeof(trs->bitbuf);
+	}
 	make_bitbuf_space(trs, len);
 	memcpy(trs->bitbuf + trs->bits_in_buf, bits, len);
 	trs->bits_in_buf += len;
@@ -96,6 +111,17 @@ int tetra_burst_sync_in(struct tetra_rx_state *trs, uint8_t *bits, unsigned int 
 			/* shift start of frame to start of bitbuf */
 			int offset = trs->next_frame_start_bitnum - trs->bitbuf_start_bitnum;
 			int bits_remaining = trs->bits_in_buf - offset;
+
+			/* guard against inconsistent bit accounting: a negative offset
+			 * (or out-of-range remainder) would make the memmove below read/
+			 * write past bitbuf[]. Resync from scratch instead. */
+			if (offset < 0 || bits_remaining < 0 ||
+			    (unsigned int)bits_remaining > sizeof(trs->bitbuf)) {
+				trs->state = RX_S_UNLOCKED;
+				trs->bitbuf_start_bitnum += trs->bits_in_buf;
+				trs->bits_in_buf = 0;
+				return len;
+			}
 
 			memmove(trs->bitbuf, trs->bitbuf+offset, bits_remaining);
 			trs->bits_in_buf = bits_remaining;

--- a/src/dsp/complex_fd.cpp
+++ b/src/dsp/complex_fd.cpp
@@ -93,6 +93,19 @@ namespace dsp {
             // Process all samples
             int outCount = 0;
             while (offset < count) {
+                // The clock-recovery PLL can diverge on noisy input (notably
+                // HackRF's 8-bit samples): a negative 'offset' makes the VOLK
+                // dot product below read before buffer[] (SIGSEGV, upstream
+                // issue #31), and a stalled 'offset' lets this loop overrun
+                // out[] -> heap corruption that trashes the decoder state
+                // (corrupted burst_cb_priv). Detect divergence and resync
+                // instead of indexing out of bounds.
+                if (offset < 0 || outCount >= STREAM_BUFFER_SIZE) {
+                    pcl.phase = 0.0f;
+                    pcl.freq = _omega;
+                    offset = count; // exit loop cleanly; offset -= count below -> 0
+                    break;
+                }
                 float error;
                 complex_t outVal;
                 complex_t dfdt;

--- a/src/dsp/osmotetra_dec.h
+++ b/src/dsp/osmotetra_dec.h
@@ -181,7 +181,13 @@ namespace dsp {
 
         inline int process(int count, const uint8_t* in, float* out)  {
             int outcnt = 0;
-            tetra_burst_sync_in(trs, (uint8_t*)in, count);
+            /* tetra_burst_sync_in()/make_bitbuf_space() corrupt the trs struct
+             * (overflowing bitbuf[4096] into burst_cb_priv) when fed more bits
+             * than the bitbuf can hold, so feed it in safe-sized chunks. */
+            const int CHUNK = 2048;
+            for (int off = 0; off < count; off += CHUNK) {
+                tetra_burst_sync_in(trs, (uint8_t*)in + off, std::min(CHUNK, count - off));
+            }
             if(out_tmp_buff.getReadable(false) > 0) {
                 outcnt += out_tmp_buff.read(out, out_tmp_buff.getReadable(false));
             }

